### ISSUE TITLE
Requirements.txt and unreasonable results

### ISSUE THREAD

**Issue by kberci** [dessa-research/fake-voice-detection#1]
Hey,
I was trying to use the model and some issues came up. 

- The first challenge I faced is that the requirements.txt file is missing, therefore the manual package installation was a little bit tricky sometimes (newest version of tensorflow does not work for instance)
- The second and bigger problem is that I tried the forward test by using the pre-trained model. I followed the instructions (copied my audio into the right folder, run the inference.py), however the prediction for both fake and real recordings were really low values. I even tried out the pre-trained model with your real voice data and it still gave only a probability of 0.3% of the real clip is being real which is supposed to be close to 100%...

1. The only error I got is: bash: _scp_spectrogram.sh: No such file or directory_
2. Also I'm not running the Atlas for trying out the pre-trained model which I'm not sure is correct or not.

Do you have any suggestion what do I miss/should install, modify to get more reasonable probabilities?
Thanks!

**Comment by ranasac19878** [comment]
Hello,

Thanks for your note:

1. Now you should see requirements.txt inside code/ folder. It was due to `.gitignore` that all the text files were ignored automatically in git push

2. We are still trying to improve our model as we speak. If you observe, the pre-trained model performs very well on the training and validation set (i.e. above 0.9 f1-score) but not so much good on test set or any other arbitrary audio file. This shows that the pre-trained model is overfitting on the algorithms that were used to produce fake voices in train and validation set. 

We would like to encourage other users to improve upon this model by performing hyperparam search to get a more generalized model. A couple of things to try are:
   a) Try to use regularization techniques
   b) Try to add more audio data to train and possibly validation set
   c) Try different model architectures (i.e. using conv1d on raw audio instead of spectrograms, or using LSTMs etc.)
   d) Analyze the pre-trained model on different audio clips to find patterns in the spectrograms where the model seems to overfit.
   c) The classes are highly imbalanced in the original dataset: 90% fake and 10% real audio clips. So, the model generally has more tendency to predict audio signals to be fake. I took some steps to overcome these by augmenting the data by using `nlpaug` library (see in `utils.py`) but it can be further improved by adding more dataset.

You should not get the ` bash: scp_spectrogram.sh: No such file or directory` error anymore since I removed that line. It was only there for some remote debugging processes

You can try pre-trained model without Atlas and that doesn't change anything.

Please see my suggestions in order to get a better model and let me know your thoughts. 

Thanks!

**Comment by Kinjal05** [comment]
Hi All,
I did not have NVIDIA GPU on my local machine. I am trying to run the code on Intel GPU but did not got success.

I wanted to confirm that can I run this code on Intel GPU or not ?

Many Thanks,
Kinjal

**Comment by Kinjal05** [comment]
Hi Team,
I am little skeptical about whether I can run the code on Windows 10 PRO version or not.
Please share your thoughts.

Thanks,
Kinjal

**Comment by ranasac19878** [comment]
Hi Kinjal, thanks for your question. Yes you should be able to run the code on Windows 10 but you might need to modify some terminal commands as the windows commands are little bit different than Mac and Linux. But the code itself should be platform independent, if you run into problems, please let us know what error you are getting.

**Comment by Kinjal05** [comment]
Thanks for your response.
However, I have managed to install Foundation Atlas in local, but facing
issue with this command: *atlas-server start*
*I am sharing trace here:*

(env) C:\Users\ram\Desktop\Foundation_Atlas>atlas-server start
2020-03-12 20:41:23,981 - Foundations Atlas Server - INFO - Starting
atlas-ce-tracker service...
2020-03-12 20:41:25,808 - Foundations Atlas Server - INFO - Started
atlas-ce-tracker service
2020-03-12 20:41:26,072 - Foundations Atlas Server - INFO - Starting
atlas-ce-archive-server service...
2020-03-12 20:41:27,493 - Foundations Atlas Server - INFO - Started
atlas-ce-archive-server service
2020-03-12 20:41:27,771 - Foundations Atlas Server - INFO - Starting
atlas-ce-rest-api service...
2020-03-12 20:41:29,431 - Foundations Atlas Server - INFO - Started
atlas-ce-rest-api service
2020-03-12 20:41:29,648 - Foundations Atlas Server - INFO - Starting
atlas-ce-gui service...
2020-03-12 20:41:31,629 - Foundations Atlas Server - INFO - Started
atlas-ce-gui service
2020-03-12 20:41:31,638 - Foundations Atlas Server - ERROR - Cannot start
atlas-ce-local-scheduler
2020-03-12 20:41:31,638 - Foundations Atlas Server - ERROR - Please try
re-installing Atlas Server via the installer
2020-03-12 20:41:31,639 - Foundations Atlas Server - INFO - Stopping
atlas-ce-gui service...
2020-03-12 20:41:42,677 - Foundations Atlas Server - INFO - Stopped
atlas-ce-gui service
2020-03-12 20:41:42,678 - Foundations Atlas Server - INFO - Removing
atlas-ce-gui service...
2020-03-12 20:41:42,726 - Foundations Atlas Server - INFO - Removed
atlas-ce-gui service
2020-03-12 20:41:42,727 - Foundations Atlas Server - INFO - Stopping
atlas-ce-rest-api service...
2020-03-12 20:41:44,611 - Foundations Atlas Server - INFO - Stopped
atlas-ce-rest-api service
2020-03-12 20:41:44,612 - Foundations Atlas Server - INFO - Removing
atlas-ce-rest-api service...
2020-03-12 20:41:44,656 - Foundations Atlas Server - INFO - Removed
atlas-ce-rest-api service
2020-03-12 20:41:44,657 - Foundations Atlas Server - INFO - Stopping
atlas-ce-archive-server service...
2020-03-12 20:41:46,257 - Foundations Atlas Server - INFO - Stopped
atlas-ce-archive-server service
2020-03-12 20:41:46,258 - Foundations Atlas Server - INFO - Removing
atlas-ce-archive-server service...
2020-03-12 20:41:46,312 - Foundations Atlas Server - INFO - Removed
atlas-ce-archive-server service
2020-03-12 20:41:46,313 - Foundations Atlas Server - INFO - Stopping
atlas-ce-tracker service...
2020-03-12 20:41:47,382 - Foundations Atlas Server - INFO - Stopped
atlas-ce-tracker service
2020-03-12 20:41:47,382 - Foundations Atlas Server - INFO - Removing
atlas-ce-tracker service...
2020-03-12 20:41:47,431 - Foundations Atlas Server - INFO - Removed
atlas-ce-tracker service
2020-03-12 20:41:47,432 - Foundations Atlas Server - INFO - Removing
foundations-atlas network service...
2020-03-12 20:41:47,768 - Foundations Atlas Server - INFO - Removed
foundations-atlas network service...

Please help for me to resolve this error.

Many Thanks,
Kinjal

**Comment by Harrshit** [comment]
Hello,

I tried to install the requirements.txt file but shows an error in the line " import numpy as np" saying no module found as "numpy" .
Kindly help me resolve this issue.

Regards,
Harshit
